pandas/tests/extension/base/methods.py

```diff
@@ -172,15 +172,16 @@ def test_combine_le(self, data_repeated):
         orig_data1, orig_data2 = data_repeated(2)
         s1 = pd.Series(orig_data1)
         s2 = pd.Series(orig_data2)
-        result = s1.combine(s2, lambda x1, x2: x1 <= x2)
+        result = s1.combine(s2, lambda x1, x2: x1 <= x2, dtype="boolean")
```

Comment:
Inferring to a different dtype than the original is quite difficult, and better to be explicit in those cases IMO.

Reply:
Although I agree in general (certainly when using `combine` for constructing the expected result), but is this change required in this specific case? (I would assume nothing changed related to this?) Because I might still be good to test the case *without* specifying the dtype explicitly, as that is public API.

Reply:
The problem here is that almost anything can be converted to a string. I've made a change though. It's inferring dtype, so not ideal performance-wise in the case where the series is a StringDtype and the result might not be.

If the user adds dtype manually, the performance is ok.